apps/prairielearn/elements/pl-drawing/elements.py

```diff
@@ -130,18 +130,18 @@ def get_attributes():
 
 
 class ControlledCurvedLine(BaseElement):
-    def generate(el, data):
-        if "draw-error-box" in el.attrib:
-            obj_draw = el.attrib["draw-error-box"] == "true"
+    def generate(self, data):
+        if "draw-error-box" in self.attrib:
+            obj_draw = self.attrib["draw-error-box"] == "true"
         else:
             obj_draw = None
 
-        offset_x = pl.get_float_attrib(el, "offset-tol-x", 0)
-        offset_y = pl.get_float_attrib(el, "offset-tol-y", 0)
-        offset_control_x = pl.get_float_attrib(el, "offset-control-tol-x", 0)
-        offset_control_y = pl.get_float_attrib(el, "offset-control-tol-y", 0)
-        grid_size = pl.get_integer_attrib(el, "grid-size", 20)
-        tol = pl.get_float_attrib(el, "tol", grid_size / 2)
+        offset_x = pl.get_float_attrib(self, "offset-tol-x", 0)
```

Comment:
I remember why I thought this was weird, the rename here is misleading here because the `el` here is actually the parsed HTML element, **NOT** an instance of this class. This is because this element is (improperly) using these methods as static methods without the decorator. @nwalters512 will defer to you on a final call for this, but I vote to revert the changes and ignore this file for linting.

Reply:
Yeah this code is pretty gross. I would still prefer to blanket-apply these rules, as they are gradually making it easier to notice these issues.

Reply:
> unless the changes to the file actually hurt readability

Isn't that precisely what's happening here? Anyone reading this would see `self` and assume that's a reference to an instance of the class, but it's not actually that. I agree with Eliot: we should not change this here. In a future PR, we can add the `@staticmethod` decorator and then apply that rule to this file, at which point it should be a no-op.

Reply:
Fair enough, I have reverted changes relating to this in 96d64b8f1.